From: Zhong LuFan <lufanzhong@gmail.com>
Subject: [PATCH] mtd: bcm47xxpart: add partition workaround for PHICOMM K3

The founder of the patch is: Tian Hao <haotia@gmail.com>

Signed-off-by: Zhong LuFan <lufanzhong@gmail.com>

--- a/drivers/mtd/bcm47xxpart.c
+++ b/drivers/mtd/bcm47xxpart.c
@@ -163,6 +163,14 @@ static int bcm47xxpart_parse(struct mtd_
 			 */
 			bcm47xxpart_add_part(&parts[curr_part++], "tplink", offset, MTD_WRITEABLE);
 			continue;
+		} else if (of_machine_is_compatible("phicomm,k3") && offset == 0x180000) {
+			/*
+			* This device has nvram_back, res_info, pro_info and dev_info from
+			* 0x180000 (end of nvram) to 0x400000 (start of linux). These partitions
+			* has essential information for original firwamre. We do not want these.
+			*/
+			bcm47xxpart_add_part(&parts[curr_part++], "phicomm", offset, MTD_WRITEABLE);
+			continue;
 		}
 
 		/* Read beginning of the block */
